### PR TITLE
[python] add test-app-python-27 app for INSTALLER_NOT_SUPPORTED_AUTO_INJECTION

### PR DESCRIPTION
**Pull request by emmettbutler**
## Motivation

We are currently not running the `INSTALLER_NOT_SUPPORTED_AUTO_INJECTION` scenario for Python 2.7. This change adds a test case for that.

## Changes

Adds a new `test-app-python-27` VM weblog

Starts running the `INSTALLER_NOT_SUPPORTED_AUTO_INJECTION` scenario with this new app

## Reviewer checklist

* [ ] If PR title starts with `[<language>]`, double-check that only `<language>` is impacted by the change
* [ ] No system-tests internal is modified. Otherwise, I have the approval from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)
* [ ] CI is green, or failing jobs are not related to this change (and you are 100% sure about this statement)
* [ ] A docker base image is modified?
    * [ ] the relevant `build-XXX-image` label is present
* [ ] A scenario is added (or removed)?
    * [ ] Get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

### CODE REVIEW

**Comment by emmettbutler** [comment]
@brettlangdon @robertomonteromiguel I think the failing tests on this PR are the result of dd-trace-py not working under py27. That's to say, the test has uncovered a real bug. Shall I mark as skipped for the moment and merge?

File: lib-injection/build/docker/python/dd-lib-python-init-test-django-27/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,12 @@
+FROM python:3.11
```
Comment (brettlangdon) [inline lib-injection/build/docker/python/dd-lib-python-init-test-django-27/Dockerfile:1]:
```suggestion
FROM python:2.7
```

?

**Review by brettlangdon — COMMENTED** [review]
```
[datadog_preload_hook] done waiting!
'import sitecustomize' failed; use -v for traceback
Traceback (most recent call last):
  File "/home/datadog/.pyenv/versions/2.7.18/lib/python2.7/site-packages/django/__main__.py", line 9, in <module>
    management.execute_from_command_line()
  File "/home/datadog/.pyenv/versions/2.7.18/lib/python2.7/site-packages/django/core/management/__init__.py", line 364, in execute_from_command_line
    utility.execute()
  File "/home/datadog/.pyenv/versions/2.7.18/lib/python2.7/site-packages/django/core/management/__init__.py", line 308, in execute
    settings.INSTALLED_APPS
  File "/home/datadog/.pyenv/versions/2.7.18/lib/python2.7/site-packages/django/conf/__init__.py", line 56, in __getattr__
    self._setup(name)
  File "/home/datadog/.pyenv/versions/2.7.18/lib/python2.7/site-packages/django/conf/__init__.py", line 41, in _setup
    self._wrapped = Settings(settings_module)
  File "/home/datadog/.pyenv/versions/2.7.18/lib/python2.7/site-packages/django/conf/__init__.py", line 110, in __init__
    mod = importlib.import_module(self.SETTINGS_MODULE)
  File "/home/datadog/.pyenv/versions/2.7.18/lib/python2.7/importlib/__init__.py", line 37, in import_module
    __import__(name)
ImportError: No module named django_app
```

> 'import sitecustomize' failed; use -v for traceback

This makes a lot of sense

> ImportError: No module named django_app

Maybe `PYTHONPATH` is treated differently for Python 2.7 and it needs an explicit `export PYTHONPATH=/home/datadog/:$PYTHONPATH`... maybe?

**Comment by emmettbutler** [comment]
@brettlangdon attempted that in https://github.com/DataDog/system-tests/pull/3455/commits/34ac1556d187a6d9641ec0f4522b505c0fcd0c58

File: utils/_context/virtual_machines.py [hunk 2]
```diff
@@ -653,7 +653,7 @@ def __init__(self, **kwargs) -> None:
             krunvm_config=None,
             os_type="linux",
             os_distro="rpm",
-            os_branch="redhat",
+            os_branch="redhat_9",
```
Comment (robertomonteromiguel) [inline utils/_context/virtual_machines.py:656]: Restore this. Currently we have disabled the RH9

**Review by robertomonteromiguel — CHANGES_REQUESTED** [review]

**Review by brettlangdon — APPROVED** [review]
minor comments/suggestions only.

File: utils/build/virtual_machine/weblogs/python/provision_test-app-python-27.yml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+            local_path: utils/build/virtual_machine/weblogs/common/test-app.service
+
+          - name: copy-service-run-script
+            local_path: utils/build/virtual_machine/weblogs/common/create_and_run_app_service.sh
+
+          - name: copy-run-weblog-script
+            local_path: utils/build/virtual_machine/weblogs/python/test-app-python-27/test-app-python_run.sh
+
+          - name: copy-python-app
+            local_path: lib-injection/build/docker/python/dd-lib-python-init-test-django-27
```
[the remark this answers is not in the document]
Comment (emmettbutler) [reply utils/build/virtual_machine/weblogs/python/provision_test-app-python-27.yml:44]: I tried that, but there's a small diff between the versions of django_app.py.

File: utils/build/virtual_machine/weblogs/python/provision_test-app-python-36.yml
Comment (brettlangdon) [inline utils/build/virtual_machine/weblogs/python/provision_test-app-python-36.yml:1]: We can remove this file, right?

File: utils/_context/virtual_machines.py [hunk 4]
```diff
@@ -700,7 +700,7 @@ def __init__(self, **kwargs) -> None:
...
```
Comment (brettlangdon) [inline utils/_context/virtual_machines.py:703]:
```suggestion
            os_branch="redhat",
```

File: utils/build/virtual_machine/weblogs/python/provision_test-app-python-27.yml [hunk 5]
```diff
@@ -0,0 +1,46 @@
+lang_variant:
+    name: Python_2.7.18
+    excluded_os_branches: [amazon_linux2, rhel_7_amd64, redhat_9]
```
Comment (brettlangdon) [inline utils/build/virtual_machine/weblogs/python/provision_test-app-python-27.yml:3]:
```suggestion
    excluded_os_branches: [amazon_linux2, rhel_7_amd64, redhat]
```

File: utils/build/virtual_machine/weblogs/python/provision_test-app-python-27.yml [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+          if ! command -v make &> /dev/null
+          then
+            sudo yum groupinstall "Development Tools" -y
+          fi
+          sudo yum -y install patch
+          sudo sh python_install.sh rpm 2.7.18
+
+weblog:
+    name: test-app-python-27
+    excluded_os_branches: [amazon_linux2, rhel_7_amd64, redhat_9]
```
Comment (brettlangdon) [inline utils/build/virtual_machine/weblogs/python/provision_test-app-python-27.yml:29]:
```suggestion
    excluded_os_branches: [amazon_linux2, rhel_7_amd64, redhat]
```

**Review by brettlangdon — APPROVED** [review]
one missed 3.6 file, and I _think_ the `redhat_9`s need to be `redhat` (if I understand Roberto's previous comments correctly), otherwise lgtm

**Review by brettlangdon — APPROVED** [review]

**Review by robertomonteromiguel — APPROVED** [review]
I only added the exclusion for ubuntu23 for this weblog and now it's working. We could merge this PR. Thanks Emmet for your work!

**Pull request merged**